packages/klighd-core/src/proxy-view/proxy-view.tsx

```diff
@@ -0,0 +1,742 @@
+        this.sizePercentage = renderOptionsRegistry.getValue(ProxyViewSize);
+        this.clusteringEnabled = renderOptionsRegistry.getValue(ProxyViewClusteringEnabled);
+        this.clusteringCascading = renderOptionsRegistry.getValue(ProxyViewClusteringCascading);
+        this.capProxyToParent = renderOptionsRegistry.getValue(ProxyViewCapProxyToParent);
+        // TODO: toggling useSynthesisProxyRendering while a proxy is on-screen doesn't get rid of the old proxy yet
+        const useSynthesisProxyRendering = renderOptionsRegistry.getValue(ProxyViewUseSynthesisProxyRendering);
+        if (this.useSynthesisProxyRendering !== useSynthesisProxyRendering) {
+            // Make sure not to use the wrong renderings if changed
+            this.clearRenderings();
+        }
+        this.useSynthesisProxyRendering = useSynthesisProxyRendering;
+        this.filterUnconnected = renderOptionsRegistry.getValue(ProxyViewFilterUnconnected);
+    }
+
+    /** Clears the {@link renderings} map. */
+    clearRenderings(): void {
+        this.renderings.clear();
+    }
+
+    /** Clears the {@link positions} map. */
+    clearPositions(): void {
+        this.positions.clear();
+    }
+
+    /**
+     * Calculates the TransformAttributes for this node's proxy, e.g. the position to place the proxy at aswell as its scale and bounds.
+     * Note that the position is pre-scaling.
+     */
+    private getTransform(node: SKNode, desiredSize: number, canvasWidth: number, canvasHeight: number, scroll: Point, zoom: number): TransformAttributes {
+        // OLD: size dependant on node's bounds
+        // const proxyWidth = size * 0.001;
+        // const proxySizeScale = Math.min(proxyHeightScale, proxyWidthScale);
+        // console.log(proxySizeScale);
+
+        const pos = this.getPosition(node);
+        const bounds = node.bounds;
+
+        // Calculate the scale and the resulting proxy dimensions
+        // The scale is calculated such that width & height are capped to a max value
+        const proxyWidthScale = desiredSize / bounds.width;
+        const proxyHeightScale = desiredSize / bounds.height;
+        const scale = Math.min(1, proxyWidthScale, proxyHeightScale);
+        const proxyWidth = bounds.width * scale;
+        const proxyHeight = bounds.height * scale;
+
+        // Center at middle of node
+        const offsetX = 0.5 * (node.bounds.width * zoom - proxyWidth);
+        const offsetY = 0.5 * (node.bounds.height * zoom - proxyHeight);
+        let x = (pos.x - scroll.x) * zoom + offsetX;
+        let y = (pos.y - scroll.y) * zoom + offsetY;
+
+        // Cap proxy at canvas border
+        x = Math.max(0, Math.min(canvasWidth - proxyWidth, x));
+        y = Math.max(0, Math.min(canvasHeight - proxyHeight, y));
+
+        // Make sure the proxies aren't rendered behind the sidebar buttons at the top right
+        /* Don't need to check for the opened sidebar since it closes as soon as the diagram is moved
```

Comment:
but if there is a proxy where the sidebar opens, should it not then adjust its position?